packages/react-reconciler/src/__tests__/ReactOffscreen-test.js

```diff
@@ -1259,4 +1261,50 @@ describe('ReactOffscreen', () => {
       </div>,
     );
   });
+
+  describe('manual interactivity', () => {
+    // @gate enableOffscreen
+    it('should attach ref only for mode null', async () => {
+      let offscreenRef;
+
+      function App({mode}) {
+        offscreenRef = useRef(null);
+        return (
+          <Offscreen
+            mode={mode}
+            ref={ref => {
+              offscreenRef.current = ref;
+            }}>
+            <div />
+          </Offscreen>
+        );
+      }
+
+      const root = ReactNoop.createRoot();
+
+      await act(async () => {
+        root.render(<App mode={'manual'} />);
+      });
+
+      expect(offscreenRef.current).not.toBeNull();
+
+      await act(async () => {
+        root.render(<App mode={'visible'} />);
+      });
+
+      expect(offscreenRef.current).toBeNull();
+
+      await act(async () => {
+        root.render(<App mode={'hidden'} />);
+      });
+
+      expect(offscreenRef.current).toBeNull();
+
+      await act(async () => {
+        root.render(<App mode={'manual'} />);
+      });
+
+      expect(offscreenRef.current).not.toBeNull();
+    });
+  });
```

Comment:
Need a test that unmounts a tree that contains an Offscreen component in manual mode, and confirms that the ref is detached.

Similarly I would add a test that has a manual Offscreen component nested inside another Offscreen component. When the outer one hides, the inner one's ref gets detached.

Reply:
Good call. You saved me a lot of hours debugging this later down the road.